### PR TITLE
Use base class name for reification and add fractional second precision to test db

### PR DESCRIPTION
**Pull request by andre1810**
- Use base class name for reification of associations
- Add explicit precision on datetime columns to enable fractional second precision on mysql

### CODE REVIEW

File: spec/dummy_app/app/models/person.rb [hunk 1]
```diff
@@ -4,12 +4,26 @@ class Person < ActiveRecord::Base
...
     belongs_to :mentor, class_name: "Person", foreign_key: :mentor_id, optional: true
   else
     belongs_to :mentor, class_name: "Person", foreign_key: :mentor_id
   end
 
+  if ActiveRecord.gem_version >= Gem::Version.new("5.0")
+    belongs_to :vehicle, class_name: "Vehicle", foreign_key: :vehicle_id, optional: true
+  else
+    belongs_to :vehicle, class_name: "Vehicle", foreign_key: :vehicle_id
+  end
```
Comment (jaredbeck) [inline spec/dummy_app/app/models/person.rb]: It seems this association (`belongs_to :vehicle`) is unused?

**Comment by andre1810** [comment]
@jaredbeck Sorry, I had to open this again due an mistake on editing the commits.

**Comment by jaredbeck** [comment]
> Sorry, I had to open this again due an mistake on editing the commits.

No worries, open as many PR as you feel like 😄 Also, check out `git rebase --interactive`, you might like it. It's pretty powerful (and dangerous).

**Comment by jaredbeck** [comment]
OK, I'm ready to merge this. Anything else you want to do?

**Comment by andre1810** [comment]
No, I´m done. Thank you 👍

**Comment by jaredbeck** [comment]
> No, I'm done. Thank you 👍

Thank you! Nice work.

> Do you have a date for the next release?

The next release from the `master` branch will probably be in April.

Please backport this by opening a PR against the `8-stable` branch, and I can do an 8.2.0 release this weekend.

**Comment by jaredbeck** [comment]
Released as 8.1.2

(I know I said 8.2.0, but I changed my mind; it's really more of a fix than an addition)

**Pull request merged**